### PR TITLE
[WIP] logdump coredns.log

### PR DESCRIPTION
**Pull request by JornShen**
/kind feature
/kind bug

see detailed dns logs in jobs presentation prow.k8s.io artifacts to help locate failed jobs problems.

ref https://github.com/kubernetes/test-infra/issues/20659

```release-note
NONE
```

### CODE REVIEW

**Comment by JornShen** [comment]
https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/98591/pull-kubernetes-e2e-gce-ubuntu-containerd/1355419454795681792/artifacts/e2e-ce839bc9f9-a7d53-minion-group-38nf/

seems that script can not dump dns log :(
Could you give some guide for it?

@wojtek-t @BenTheElder

File: cluster/log-dump/log-dump.sh [hunk 1]
```diff
@@ -42,7 +42,7 @@ readonly node_ssh_supported_providers="gce gke aws"
 readonly gcloud_supported_providers="gce gke"
 
 readonly master_logfiles="kube-apiserver.log kube-apiserver-audit.log kube-scheduler.log kube-controller-manager.log etcd.log etcd-events.log glbc.log cluster-autoscaler.log kube-addon-manager.log konnectivity-server.log fluentd.log kubelet.cov"
-readonly node_logfiles="kube-proxy.log fluentd.log node-problem-detector.log kubelet.cov"
+readonly node_logfiles="kube-proxy.log fluentd.log node-problem-detector.log kubelet.cov coredns.log"
```
Comment (BenTheElder) [inline cluster/log-dump/log-dump.sh:45]: What reason do you have to think this is the name of the log?

Comment (BenTheElder) [reply cluster/log-dump/log-dump.sh:45]: Also: the other logs listed here will exist on *every* node. The in cluster DNS is N pods, not a daemonset

Comment (JornShen) [reply cluster/log-dump/log-dump.sh:45]: yeah. So we use `kubectl logs` to logdump coredns logs ? @BenTheElder

Comment (mm4tt) [reply cluster/log-dump/log-dump.sh:45]: This script copies logs from the `/var/log` directory. Adding coredns.log here assumes that there is a `/var/log/coredns.log` present on a node. I believe it's not a correct assumption. To make it work you should change to coredns addon to write logs to `/var/log`.

The other thing is that your are making the change in the deprecated directory. See the README.md file in the log-dump dir. You should make the change in the test-infra repo - https://github.com/kubernetes/test-infra/blob/6505d464406ee67c25fa67ec75de802cb231a753/logexporter/cluster/log-dump.sh

@tosi3k ^

@BenTheElder, it's fine if the file is not present on all nodes. In the scale tests, we already dump "cl2-*" (via setting the `LOG_DUMP_EXTRA_FILES` env) logs that are not present on every node.

**Comment by JornShen** [comment]
It is not suitable to add here.  Thanks for @mm4tt @BenTheElder guiding. 
/close

**Pull request closed without merging**